jib-plugins-common/src/test/java/com/google/cloud/tools/jib/plugins/common/UpdateCheckerTest.java

```diff
@@ -0,0 +1,201 @@
+    Instant before = Instant.now();
+    setupConfigAndLastUpdateCheck();
+    Optional<String> message =
+        UpdateChecker.performUpdateCheck(false, "2.0.0", testWebServer.getEndpoint(), configDir);
+    Assert.assertFalse(message.isPresent());
+    String modifiedTime =
+        new String(
+            Files.readAllBytes(configDir.resolve("lastUpdateCheck")), StandardCharsets.UTF_8);
+    Assert.assertTrue(Instant.parse(modifiedTime).isAfter(before));
+  }
+
+  @Test
+  public void testPerformUpdateCheck_noConfigOrLastUpdateCheck() throws IOException {
+    Instant before = Instant.now();
+    Optional<String> message =
+        UpdateChecker.performUpdateCheck(false, "1.0.2", testWebServer.getEndpoint(), configDir);
+    Assert.assertTrue(message.isPresent());
+    Assert.assertEquals(
+        "A new version of Jib (2.0.0) is available (currently using 1.0.2). Update your build "
+            + "configuration to use the latest features and fixes!",
+        message.get());
+    String modifiedTime =
+        new String(
+            Files.readAllBytes(configDir.resolve("lastUpdateCheck")), StandardCharsets.UTF_8);
+    Assert.assertTrue(Instant.parse(modifiedTime).isAfter(before));
+  }
+
+  @Test
+  public void testPerformUpdateCheck_lastUpdateCheckTooSoon() throws IOException {
+    FileTime modifiedTime = FileTime.from(Instant.now().minusSeconds(12));
+    setupConfigAndLastUpdateCheck();
+    Files.write(
+        configDir.resolve("lastUpdateCheck"),
+        modifiedTime.toString().getBytes(StandardCharsets.UTF_8));
+    Files.setLastModifiedTime(configDir.resolve("lastUpdateCheck"), modifiedTime);
+    Optional<String> message =
+        UpdateChecker.performUpdateCheck(false, "1.0.2", testWebServer.getEndpoint(), configDir);
+    Assert.assertFalse(message.isPresent());
+
+    // lastUpdateCheck should not have changed
+    String lastUpdateTime =
+        new String(
+            Files.readAllBytes(configDir.resolve("lastUpdateCheck")), StandardCharsets.UTF_8);
+    Assert.assertEquals(Instant.parse(lastUpdateTime), modifiedTime.toInstant());
+  }
+
+  @Test
+  public void testPerformUpdateCheck_skip() {
+    Optional<String> message =
+        UpdateChecker.performUpdateCheck(true, "1.0.2", testWebServer.getEndpoint(), configDir);
+    Assert.assertFalse(message.isPresent());
+  }
+
+  @Test
+  public void testPerformUpdateCheck_systemProperty() {
+    System.setProperty(PropertyNames.DISABLE_UPDATE_CHECKS, "true");
+    Optional<String> message =
+        UpdateChecker.performUpdateCheck(true, "1.0.2", testWebServer.getEndpoint(), configDir);
+    Assert.assertFalse(message.isPresent());
+  }
+
+  @Test
+  public void testPerformUpdateCheck_configDisabled() throws IOException {
+    UpdateChecker.ConfigJsonTemplate config = new UpdateChecker.ConfigJsonTemplate();
+    config.setDisableUpdateCheck(true);
+    JsonTemplateMapper.writeTo(config, Files.newOutputStream(configDir.resolve("config.json")));
```

Comment:
I think it's nice to write an actual JSON, like `Files.write(..., "{\"disableUpdateCheck\":true}".getBytes(...))`.